### PR TITLE
adds agent orange to CAS so you can recreate Vietnam, it removes plants/weeds/resin and fills the area it hits with horrible debilitating smoke.

### PR DESCRIPTION
**Pull request by F-OS**
- [x] Compiles.

- [x] Does not runtime on use.
- [x] Smoke causes damage.
- [x] Launchable from DS.

- [x] Ensure it properly manufactures.

- [ ] Adjust sprite.

- [x] Add intube sprite

- [ ] Adjust for balance

- [ ] clean up code.

Why?: I feel giving the dropship more area denial utilities may improve gameplay a bit.

Pros: Adds more dropship utility
Cons: May fuck with the game flow by denying areas to both sides for a while.

### CODE REVIEW

**Comment by MoXuT0** [comment]
After some testing it appears resin holes don't get destroyed by gas and remains after even with huggers inside.

**Comment by F-OS** [comment]
I almost forgot this existed! Lemme fix it in a sec.

File: code/modules/cm_marines/dropship_ammo.dm [hunk 1]
```diff
@@ -369,7 +410,7 @@
 			S.set_up(1,0,T,null)
 			S.start()
 		spawn(10)
-			new/obj/item/device/flashlight/flare/on/cas(T) 
+			new/obj/item/device/flashlight/flare/on/cas(T)
```
Comment (Rohesie) [inline code/modules/cm_marines/dropship_ammo.dm]: Shouldn't this be `new /obj/item/device/flashlight/flare/on/cas(T)`?

**Comment by Rohesie** [comment]
Do you want to change anything else in it?

**Comment by F-OS** [comment]
i have some plans for this in a bit.

File: code/game/objects/effects/effect_system/smoke.dm [hunk 2]
```diff
@@ -167,6 +167,47 @@
...
 
+/////////////////////////////////////////////
+// Agent Orange
+/////////////////////////////////////////////
+/obj/effect/particle_effect/smoke/agentorange
+	name = "agent orange smoke"
+	color = "#ff3b00"
+	time_to_live = 60 //hopefully this is about a minute
+/obj/effect/particle_effect/smoke/agentorange/Move()
+	..()
```
Comment (Qustinnus) [inline code/game/objects/effects/effect_system/smoke.dm]: .=..()

File: code/game/objects/effects/effect_system/smoke.dm [hunk 3]
```diff
@@ -167,6 +167,47 @@
...
+	for(var/mob/living/carbon/human/R in get_turf(src))
+		affect(R)
+
+/obj/effect/particle_effect/smoke/agentorange/affect(var/mob/living/carbon/R)
+	..()
```
Comment (Qustinnus) [inline code/game/objects/effects/effect_system/smoke.dm]: .=..()

File: code/game/objects/effects/effect_system/smoke.dm [hunk 4]
```diff
@@ -167,6 +167,47 @@
...
+	var/antispam
+	if(isXeno(R))
+		R.adjustFireLoss(R.maxHealth*(rand(0.02, 0.1))) // Eats through xeno health like candy because they're made of resin or it has a reaction with plasma or some peusdoscience bullshit
+	else if(ishuman(R))
+		var/mob/living/carbon/human/H = R
+		if(istype(H.wear_mask, /obj/item/clothing/mask/gas))
```
Comment (Qustinnus) [inline code/game/objects/effects/effect_system/smoke.dm]: not really the best way to do this but Im unsure if CM has some form of bitflag for specific protection types such as breathing filtering or skin protection

File: code/game/objects/effects/effect_system/smoke.dm [hunk 5]
```diff
@@ -167,6 +167,47 @@
...
+			H.apply_damages(0, 1, 0) // it's still touching your skin soooooo
+		else
+			H.adjustToxLoss(3) // If you don't wear a gas mask you get uberfucked.
+			if(antispam != 1)
```
Comment (Qustinnus) [inline code/game/objects/effects/effect_system/smoke.dm]: if(!antispam)

File: code/game/objects/effects/effect_system/smoke.dm [hunk 6]
```diff
@@ -167,6 +167,47 @@
...
+				antispam = 1
```
Comment (Qustinnus) [inline code/game/objects/effects/effect_system/smoke.dm]: antispam = TRUE

File: code/game/objects/effects/effect_system/smoke.dm [hunk 7]
```diff
@@ -167,6 +167,47 @@
...
+				if(ishuman(R)) //Humans only to avoid issues
```
Comment (Qustinnus) [inline code/game/objects/effects/effect_system/smoke.dm]: you already checked if they're human

Comment (Qustinnus) [reply code/game/objects/effects/effect_system/smoke.dm]: you havnt even been using the R variable for a while up until here

File: code/game/objects/effects/effect_system/smoke.dm [hunk 8]
```diff
@@ -167,6 +167,47 @@
...
+					H.setHalLoss(20) // PAIN
+					if(prob(20))
+						H.eye_blurry += rand(12,20)
+					if(prob(15))
+						to_chat(H, "<span class='danger'>OH GOD, YOUR ENTIRE BODY BURNS!</span>")
+						H.vomit() // im evil
+					if(prob(10))
+						H.apply_damages(0, 4, 0)
+				spawn(10)
```
Comment (Qustinnus) [inline code/game/objects/effects/effect_system/smoke.dm]: this is not how you should do cooldowns. look at a piece of cooldown code for this (one that uses world.time)

File: code/modules/cm_marines/dropship_ammo.dm [hunk 9]
```diff
@@ -228,6 +228,47 @@
...
+
+/obj/structure/ship_ammo/rocket/agentorange
+	name = "\improper DMC-1207 'CreepClean'"
+	desc = "The DMC-1207 is a modification of the AGM-227 that has had it's explosive payload replaced with a canister of pressurized nitrogen sorrunded with defoliants and toxins, meant for cleaning out large areas during asymetrical warfare to deny cover to the enemy."
+	icon_state = "singlegas"
+	travelling_time = 30 //not powerful, but reaches target fast
+	ammo_id = ""
+	point_cost = 300
+
+	detonate_on(turf/impact)
```
Comment (Qustinnus) [inline code/modules/cm_marines/dropship_ammo.dm]: no relative pathing

**Comment by F-OS** [comment]
Closing because i'm too tired to even begin trying to fix all the little bugs in here and I don't want to clutter the open PRs.

**Pull request closed without merging**